nova_vm/src/ecmascript/builtins/control_abstraction_objects/promise_objects/promise_prototype.rs

```diff
@@ -34,16 +50,73 @@ impl Builtin for PromisePrototypeThen {
 }
 
 impl PromisePrototype {
-    fn catch(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    fn catch(agent: &mut Agent, this_value: Value, args: ArgumentsList) -> JsResult<Value> {
+        // 1. Let promise be the this value.
+        // 2. Return ? Invoke(promise, "then", « undefined, onRejected »).
+        let property_key = PropertyKey::from_static_str(agent, "then");
+        let on_rejected = args.get(0);
+        if let Value::Promise(promise) = this_value {
+            let then_is_overriden = match agent[promise].object_index {
```

Comment:
**issue:** It's also possible that then is overridden at the Promise prototype level, as stupid as that is :(

Reply:
Properly handling this turned out to be more complicated than anticipated, so we're instead making everything take the slow path in this PR. We'll revisit the fast path in a follow-up.